lib/ansible/modules/network/edgeos/edgeos_config.py

```diff
@@ -206,67 +242,118 @@ def check_command(module, command):
 
 
 def diff_config(module, commands, config):
+    """Diff the candidate commands against current config returning a list of
+    updates to be applied to remote edgeos device
+
+    :param module: ansible module for this type (edgeos)
+    :type module: ansible.module
+    :param commands: candidate commands passed through ansible
+    :type commands: list
+    :param config: [commands pulled from edgeos device]
+    :type config: list
+    :return: updates: changes to apply to remote device
+    :rtype: list
+    :return: unmanaged_config: config on device without matching candidate
+    commands passed to ansible
+    :rtype: list
+    :return: invalid: commands passed to ansible not starting with 'set' or
+    'delete' and therefore considered invalid
+    :rtype: list
+    """
     config = [to_native(check_command(module, c)) for c in config.splitlines()]
 
-    updates = list()
-    visited = set()
-    delete_commands = [line for line in commands if line.startswith('delete')]
+    set_commands, delete_commands, invalid_commands = list(), list(), list()
+    updates, unmanaged_config = list(), list()
```

Comment:
Use literals rather than constructors.
```suggestion
    set_commands, delete_commands, invalid_commands = [], [], []
    updates, unmanaged_config = [], []
```